src/main.rs

```diff
@@ -246,21 +248,17 @@ fn serialize_ways(
     Ok(stats)
 }
 
-fn build_relations_index<I, FileCreator>(
-    create_file: FileCreator,
-    block_index: I,
-) -> Result<ids::IdTable, Error>
+fn build_relations_index<I>(data: Arc<Mmap>, block_index: I) -> Result<ids::IdTable, Error>
```

Comment:
How about just using &[u8] everywhere instead of the Mmap type?

Reply:
I was thinking about this. However, `parallel_process` needs producer to have a static lifetime if I am not mistaken.

Reply:
Refactored `parallel_process` with `crossbeam::scope` will allows to use references from stack. Note: `rayon::scope` does not work here, since it requires consumer to be Send, which we cannot do in our case since flatdata uses `Rc` everywhere.